File: 240504/홀수 짝수의 묶음/odd-even-bundle.py
n = int(input())
ans = 0

odd = 0
even = 0

numbers = list(map(int, input().split()))

for num in numbers:
    if num % 2 != 0:
        odd += 1
    else:
        even += 1

if even > odd:
    ans = odd * 2 + 1
elif even == odd:
    ans = even + odd
else:
    ans = even * 2
    size = odd - even

    if size % 3 == 0:
        ans += (size // 3) * 2
    else:
        if (size % 3) % 2 == 0:
            ans += size // 3 * 2 + 1
        else:
            ans += size // 3 * 2 - 1

print(ans)

File: 260501/홀수 짝수의 묶음/odd-even-bundle.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    // Please write your code here.
    int ans = 0, odd = 0, even = 0;
    int N;
    cin >> N;
    vector<int> numbers(N);
    for(int i = 0; i < N; i++)
        cin >> numbers[i];

    for(int i = 0; i < N; i++)
    {
        if(numbers[i] % 2 != 0)
            odd += 1;
        else
            even += 1;
    }

    if(even > odd)
        ans = odd * 2 + 1;
    else if(even == odd)
        ans = even + odd;
    else
    {
        ans = even * 2;
        int size = odd - even;

        if(size % 3 == 0)
            ans += (int)(size / 3) * 2;
        else
        {
            if((size % 3) % 2 == 0)
                ans += (int)(size / 3) * 2 + 1;
            else
                ans += (int)(size / 3) * 2 - 1;
        }
    }
    cout << ans;
    return 0;
}
